Source/SwiftLintFramework/Rules/Lint/PrivateUnitTestRule.swift

```diff
@@ -156,6 +192,77 @@ private class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
     }
 }
 
+private class Rewriter: SyntaxRewriter, ViolationsSyntaxRewriter {
+    private(set) var correctionPositions: [AbsolutePosition] = []
+    let locationConverter: SourceLocationConverter
+    let disabledRegions: [SourceRange]
+    private let parentClassRegex: NSRegularExpression
+
+    init(parentClassRegex: NSRegularExpression,
+         locationConverter: SourceLocationConverter,
+         disabledRegions: [SourceRange]) {
+        self.locationConverter = locationConverter
+        self.disabledRegions = disabledRegions
+        self.parentClassRegex = parentClassRegex
+    }
+
+    override func visit(_ node: ClassDeclSyntax) -> DeclSyntax {
+        guard node.hasParent(matching: parentClassRegex), node.isPrivate else {
```

Comment:
Nit: reverse condition order since I expect `node.isPrivate` to be cheaper than matching the parent class regex.

```suggestion
        guard node.isPrivate, node.hasParent(matching: parentClassRegex) else {
```

Reply:
Good point!